at-luantran/Week4/2018_07_10_exercise3_rubyday2.rb

```diff
@@ -0,0 +1,24 @@
+class ArrayClone
+  def initialize
+    @array = [1, 2, 4]
+    @length = 0
+  end
+
+  def map
+    arr_out = []
+    @array.each { |x| arr_out.push(yield(x)) }
+    arr_out
+  end
+
+  def select
+    arr_out = []
+    @array.each do |x|
+      arr_out.push(x) if yield(x)
+    end
+    arr_out
+  end
+end
+
+arr_clone = ArrayClone.new
+p arr_clone.select(&:even?)
+p arr_clone.map { |x| x + 1 }
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/a57ae48d92fb821ec476e758eb39ed7da22c6715/at-luantran/Week4/2018_07_10_exercise3_rubyday2.rb#L24"></span>Lint/AmbiguousBlockAssociation: Parenthesize the param <code>arr_clone.map { |x| x + 1 }</code> to make sure that the block will be associated with the <code>arr_clone.map</code> method call. (https://github.com/rubocop-hq/ruby-style-guide#syntax)</td>
    </tr>
  </tbody>
</table>